src/test/java/com/rtb/projectmanagementtool/auth/AuthOpsTest.java

```diff
@@ -54,18 +43,35 @@ public void tearDown() {
 
   @Test
   public void testLoginUser() {
+    // add list of mock users
+    ArrayList<UserData> testUsers = new ArrayList<UserData>();
+    testUsers.add(new UserData(1l, "abc"));
+    testUsers.add(new UserData(2l, "opq"));
+    testUsers.add(new UserData(3l, "xyz"));
+
     // new controller
     UserController controller = mock(UserController.class);
-    controller.addUser(testUser);
 
     // build and send a test cookie
-    cookie = mock(Cookie.class);
-    cookie.setName("sessionUserID");
-    cookie.setValue("out");
+    Cookie[] testCookies = new Cookie[3];
+    testCookies[0] = new Cookie("SessionUserID", "abc");
+    testCookies[1] = new Cookie("SessionUserID", "zvm");
+    testCookies[2] = new Cookie("SessionUserID", "wpk");
+
+    // on this call in class method, return test user
+    when(controller.getEveryUser()).thenReturn(testUsers);
+
+    // when get cookies is called, pass cookies
+    when(request.getCookies()).thenReturn(testCookies);
 
     // call loginUser
-    loginUser(request, response);
+    auth.loginUser(request, response);
+
+    // captor setup
+    ArgumentCaptor<Cookie> cookieCaptor = ArgumentCaptor.forClass(Cookie.class);
+    verify(response).addCookie(cookieCaptor.capture());
```

Comment:
Wanted but not invoked:
httpServletResponse.addCookie(
    <Capturing argument>
);

Reply:
I think I'm not understanding exactly what's going on here. I'm trying to get the cookies from the response (ie: response.getCookies(), but this way because that method doesn't exist. This line (72), I think is implementing a way to "get" the cookies from the response. Yet, there are zero interactions with this mock.

Reply:
starting off with the "wanted but not invoked" error, that means we're expecting the test to call addCookie but it doesn't get called.  Most probably that [line in authHelpers](https://github.com/googleinterns/step171-2020/pull/46/files/6bf3c7946b97740c326ac923f70145bf118fe057#diff-975ee4eb190e78d0a053b7d66bafe0abR60) is not getting reached.  Maybe add a couple print statements in there to see why it's not getting to that last bit.

Reply:
lines 71-75 I think is good